ingest-app/src/cmr/ingest/api/subscriptions.clj

```diff
@@ -60,6 +62,28 @@
            subscriber-id
            concept-id)))))
 
+(defn- search-gran-refs-with-sub-params
+  "Performs a granule search using the provided query parameters. If the query is no good,
+   we throw a service error."
+  [context params]
+  (try
+    (search/find-granule-references context params)
+    (catch Exception e
+      (errors/throw-service-error :bad-request
+       (str "Error ingesting subscription: invalid query parameters [" (dissoc params :token) "]")))))
+
+(defn- validate-query
+  "Performs a granule search using subscription query parameters for purposes of validation"
+  [request-context subscription]
+  (let [metadata (-> (:metadata subscription) (json/decode true))
+        collection-id (:CollectionConceptId metadata)
+        query-string (:Query metadata)
+        query-params (sub-common/create-query-params query-string)
+        search-params (merge {:collection-concept-id collection-id
```

Comment:
add a `:page-size 0` to limit the request size.

Reply:
Now this is smart.

Reply:
Oh!! I deleted the wrong comment, and I don't know how to get it back.  Sorry Mitch - he stated: "now this is smart." and I agree with him.